### PR TITLE
Add wrapped mode support

### PR DESCRIPTION
**Pull request by und3f**
UPD1: Add wrapped snake support

### CODE REVIEW

**Comment by und3f** [comment]
The PR is ready for the review.

Example wrapped game board
image

File: src/lib/components/SvgSnakeBody.svelte [hunk 1]
```diff
@@ -26,153 +33,218 @@
       body.pop();
     }
 
+    // Split wrapped body parts into separated segments
+    const bodySegments = ((body: Point[]): Point[][] => {
```
Comment (coreyja) [inline src/lib/components/SvgSnakeBody.svelte:37]: Might be nice to extract nicely named function for this. Maybe `splitBodySegments`?

I didn't see at first that this was a self-evaluating function

File: src/lib/components/SvgSnakeBody.svelte [hunk 2]
```diff
@@ -26,153 +33,218 @@
...
-    if (bodyCenterPoints.length > 0) {
-      // Use overlap to ensure that we connect to head and tail.
+    if (body.length > 0) {
       const gapSize = svgCalcParams.cellSpacing + OVERLAP;
 
+      // Extend each wrapped segment endings towards border
+      for (let i = 0; i < bodySegmentsCenterPoints.length; i++) {
+        // Extend each segment last point towards border
+        if (i < bodySegmentsCenterPoints.length - 1) {
+          const cur = bodySegmentsCenterPoints[i].slice(-1)[0];
```
Comment (coreyja) [inline src/lib/components/SvgSnakeBody.svelte:82]: NitPick: We can probably use `at(-1)` instead of the `slice(-1)[0]` but 🤷

**Review by coreyja — COMMENTED** [review]
This looks great!

I left two code level comments but that don't affect functionality at all!

I wasn't able to pull this down and run it [just on the wrong machine now] but I'll try to do that shortly and give it the final 👍

File: src/lib/components/SvgSnakeBody.svelte (hunk 1 above)
Comment (und3f) [reply src/lib/components/SvgSnakeBody.svelte]: Applied

File: src/lib/components/SvgSnakeBody.svelte (hunk 2 above)
Comment (und3f) [reply src/lib/components/SvgSnakeBody.svelte]: As we discussed it online refactored obtaining last items with util function `last(arr)`

**Comment by und3f** [comment]
Added version with `at()`

**Review by coreyja — APPROVED** [review]
This is looking good to me! I'll work on getting this deployed this weekend! 💯

**Pull request merged**